src/huntsman/pocs/utils/pyro/service.py

```diff
@@ -37,6 +34,10 @@ def pyro_service(service_class=None,
             service was started via `autostart=True`, the function will block until terminated,
             but still return the completed process.
     """
+    while not server_is_running():
+        logger.debug("Waiting for config server to start up before starting pyro service.")
+        time.sleep(5)
+
```

Comment:
This can hang forever.

I made the suggestion to use `wait-for-it` before the service starts, rather than part of the service itself.

https://github.com/AstroHuntsman/huntsman-pocs/issues/290#issuecomment-758152699

Reply:
@wtgee Thanks for the comment. Yes, it can hang forever, but I sort of want it to if the config server isn't running. Not sure what the advantages are of using `wait-for-it`? I'm trying to keep the shell scripts as minimal as possible, and we probably want to wait for the config server before starting pyro services in general.